pkg/secrets/clients/conjur/conjur_secrets_retriever.go

```diff
@@ -60,6 +60,10 @@ func (retriever conjurSecretRetriever) Retrieve(variableIDs []string) (map[strin
 func retrieveConjurSecrets(accessToken []byte, variableIDs []string) (map[string][]byte, error) {
 	log.Info(messages.CSPFK003I, variableIDs)
 
+	if len(variableIDs) == 0 {
+		return nil, nil
+	}
```

Comment:
I might be missing some context, but wouldn't Secrets Provider not retrieving any secrets from Conjur be a condition where failure is expected and preferable?

If there is some context where this is an acceptable outcome, should we at least provide an Info or Warn level log when Secrets Provider isn't retrieving any secrets? I think the log messaging would be a bit unclear.

Example: if no error is returned at this point, then eventually the SP logs will display a success message like `Secrets [pushed to file/updated in K8s] successfully` when actually no secrets were pushed/updated.

Reply:
I've updated the code to produce an INFO level message when this occurs. I'm open for debate on whether this should be an error, but my preference is to leave this as info for future corner cases.

Reply:
Thinking about this some more, if we did want things to error out for this sort of misconfiguration, I think a better place for it would be to check within each group, maybe even making it conditional on secret file format for each group. Said another way, even if we did "error out" here when no Conjur secrets are retrieved for all groups, there could still be a case where one particular secret group has not specified any secrets to retrieve from Conjur, but other secret groups have.

Reply:
> I think the best bet is to add a warning here, but otherwise allow for zero secrets ... for "future proofing"

I agree! Thanks for context around reasonable no-secret scenarios.

> I think a better place for [errors re. zero secrets] would be to check within each group

Definitely a better idea to approach the no-secret scenarios on a per-group basis. That said, I'm all for the INFO level log ATM.